ENIGMAsystem/SHELL/Bridges/General/Win32-Direct3D11.cpp

```diff
@@ -309,65 +148,55 @@ namespace enigma
     depthStencilDesc.BackFace.StencilPassOp = D3D11_STENCIL_OP_KEEP;
     depthStencilDesc.BackFace.StencilFunc = D3D11_COMPARISON_ALWAYS;
 
-    // Create the depth stencil state.
     result = m_device->CreateDepthStencilState(&depthStencilDesc, &m_depthStencilState);
-    if(FAILED(result))
-    {
+    if (FAILED(result)) {
       //return false;
     }
 
-    // Set the depth stencil state.
     m_deviceContext->OMSetDepthStencilState(m_depthStencilState, 1);
 
-    // Initailze the depth stencil view.
-    ZeroMemory(&depthStencilViewDesc, sizeof(depthStencilViewDesc));
-
     // Set up the depth stencil view description.
+    D3D11_DEPTH_STENCIL_VIEW_DESC depthStencilViewDesc = { };
     depthStencilViewDesc.Format = DXGI_FORMAT_D24_UNORM_S8_UINT;
     depthStencilViewDesc.ViewDimension = D3D11_DSV_DIMENSION_TEXTURE2D;
     depthStencilViewDesc.Texture2D.MipSlice = 0;
 
-    // Create the depth stencil view.
     result = m_device->CreateDepthStencilView(m_depthStencilBuffer, &depthStencilViewDesc, &m_depthStencilView);
-    if(FAILED(result))
-    {
+    if (FAILED(result)) {
       //return false;
     }
 
-    // Bind the render target view and depth stencil buffer to the output render pipeline.
     m_deviceContext->OMSetRenderTargets(1, &m_renderTargetView, m_depthStencilView);
 
     // Setup the raster description which will determine how and what polygons will be drawn.
+    D3D11_RASTERIZER_DESC rasterDesc;
     rasterDesc.AntialiasedLineEnable = false;
-    rasterDesc.CullMode = D3D11_CULL_BACK;
+    rasterDesc.CullMode = D3D11_CULL_NONE;
     rasterDesc.DepthBias = 0;
     rasterDesc.DepthBiasClamp = 0.0f;
-    rasterDesc.DepthClipEnable = true;
+    rasterDesc.DepthClipEnable = false;
     rasterDesc.FillMode = D3D11_FILL_SOLID;
     rasterDesc.FrontCounterClockwise = false;
     rasterDesc.MultisampleEnable = false;
     rasterDesc.ScissorEnable = false;
     rasterDesc.SlopeScaledDepthBias = 0.0f;
 
-    // Create the rasterizer state from the description we just filled out.
     result = m_device->CreateRasterizerState(&rasterDesc, &m_rasterState);
-    if(FAILED(result))
-    {
+    if (FAILED(result)) {
```

Comment:
Why bother keeping this?

Reply:
I don't know, but I do check similar results in the Direct3D9 bridge to see if it's initialized correctly. I also see I have a bunch here, perhaps more than necessary, as a side effect of Direct3D11's API. I should probably add real error messages. Do you think I should wrap these checks in debug macros?

Reply:
Ok, so I added `show_error` inside all of the failed checks for now since you requested I use that elsewhere already. This leaves me with several questions I wish you'd answer.

1) With `show_error` that is fatal, do I need to add the return explicitly?
2) Should all of these be fatal errors? I believe they should because in all of the cases you really can't continue or draw anything.
3) Do you want me to only check these results in debug mode or can I do it in run mode too? This is a kind of important area to check them that I believe GM also does in run/compile mode because end-users may run into these if the DX runtime is not installed/etc.

Please let me know your thoughts on these questions.

Reply:
Yeah, no graphics can be rendered if you fail to create the device/swap chain. Since fatal `show_error` does not return, should I make it return? Why can't we make fatal `show_error` call `std::abort`?

Reply:
A fatal `show_error` *does* call `abort()`; that's why it doesn't return. Actually, I think it might call `exit(1)` or something. But you get the idea.

Reply:
Ok, that's good then, I'm satisfied with this.